scipy/cluster/hierarchy.py

```diff
@@ -2330,7 +2330,8 @@ def num_obs_linkage(Z):
     12
 
     """
-    Z = as_xparray(Z, order='c')
+    xp = array_namespace(Z)
+    Z = _asarray(Z, order='c', xp=xp)
```

Comment:
The only benefit of making this a one-liner is LoC - the downsides are that it is less explicit, and the LoC will return if we are able to use `xp.asarray` anyway. Furthermore, if the function was changed to later use `xp`, this change would have to be made (else have duplicate `array_namespace` calls).